xdsl/dialects/scf.py

```diff
@@ -536,6 +536,45 @@ def __init__(
     def get(cond: SSAValue | Operation, *output_ops: SSAValue | Operation) -> Condition:
         return Condition(cond, *output_ops)
 
+    def print(self, printer: Printer):
+        printer.print("(", self.cond, ") ")
+        if self.attributes:
+            printer.print_op_attributes(self.attributes)
+            printer.print(" ")
+        if self.arguments:
+            printer.print_list(self.arguments, printer.print_ssa_value)
+            printer.print_string(" : ")
+            printer.print_list(
+                self.arguments, lambda val: printer.print_attribute(val.type)
+            )
+
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        parser.parse_punctuation("(")
+        unresolved_cond = parser.parse_unresolved_operand("cond expected")
+        parser.parse_punctuation(")")
+        cond = parser.resolve_operand(unresolved_cond, IntegerType(1))
+        attrs = parser.parse_optional_attr_dict()
+
+        # The optional list of ssa values can be confused with the list of results of the
+        # next operation. We save the position to backtrack in case we can't parse the
+        # types.
+        pos = parser.pos
+        unresolved_arguments = parser.parse_optional_undelimited_comma_separated_list(
+            parser.parse_optional_unresolved_operand, parser.parse_unresolved_operand
+        )
+        if unresolved_arguments is not None and parser.parse_optional_punctuation(":"):
```

Comment:
This will parse `%arg0, %arg1` and discard it if it doesn't find the `:`.
Instead, you should just have a `parse_punctuation(':')` in the `if` directly, and just do the condition on `unresolved_arguments`.